EnvChanger/Classes/EnvChangerController.swift

```diff
@@ -32,45 +32,103 @@ public class EnvChangerController<T>: UIViewController where T: EnvironmentRepre
         var selectedEnv: T?
     }
     
-    var envs: T.Type!
-    var completionHandler: ((T) -> Void)!
+    public var savedEnvironment: String? {
+        userDefaults.string(forKey: activeEnvKey)
+    }
     
-    private let ACTIVE_ENV_KEY = "ENV_CHANGER_CURRENT_SAVED_ENVIRONMENT"
+    private let activeEnvKey = "ENV_CHANGER_CURRENT_SAVED_ENVIRONMENT"
     private let window = EnvChangerWindow()
+    private let appWindow: UIWindow?
+    private let userDefaults = UserDefaults.standard
+    private let application = UIApplication.shared
```

Comment:
I don't like these, they are just convenience accessors, use computed property instead of static initialisation of the value

Reply:
Changed to computed properties.